pallets/providers/src/utils.rs

```diff
@@ -49,10 +50,22 @@ impl<T> Pallet<T>
 where
     T: pallet::Config,
 {
-    pub fn do_msp_sign_up(who: &T::AccountId, msp_info: &MainStorageProvider<T>) -> DispatchResult {
+    /// This function tests the randomness of the pallet. TODO: remove this, only for testing
+    pub fn test_randomness_output(who: &T::AccountId) -> (T::Hash, BlockNumberFor<T>) {
+        T::ProvidersRandomness::random(who.encode().as_ref())
+    }
+
+    /// This function holds the logic that checks if a user can request to sign up as a Main Storage Provider
+    /// and, if so, stores the request in the SignUpRequests mapping
+    pub fn do_request_msp_sign_up(
```

Comment:
You should check if the account does not already have a pending sign up request in `SignUpRequests` storage.

Reply:
My idea was that if a user has a pending sign up request and is trying to request another sign up, we just overwrite the previous one with the new one, but now that I'm thinking we would have to add the logic to check for it and change the held amount, so I think your solution is more elegant and simple. 👍🏻

Reply:
This actually exposes a kind of vulnerability. We do not allow for a same account to be a BSP and MSP at the same time. But you can sign up for both while not being confirmed, and then confirm them, becoming both. All the more reason to not allow SPs to re-sign up when they have a request ongoing.

Reply:
Ya like @ffarall points out, it leads to many caveats.

Reply:
Mmmm not really, since the storage that holds the requests is a mapping from an account to a single request. That means that an account can only have one sign up request active, so the user wouldn't be able to register as both a BSP and a MSP.
Still, I agree that it complicates the logic quite a bit just to allow users to not have to make an extra call (to `cancel_sign_up`) so it's not really worth it

Reply:
Good point, still checking I think is safer.